### DISCUSSION TITLE
Improvement of coverage for FFMPEG supported formats

### DISCUSSION THREAD

**Discussion by kimmot** [nschlia/ffmpegfs#108] [General]
First of all thank you very much for the great work. I tested ffmpegfs to implement a MP3 mirror of the music library in Daphile (www.daphile.com). It seems to do exactly what I hoped for.

I noticed however that it didn't support some audio formats that are still supported by ffmpeg e.g. DSF, DFF and APE. Also some file extension (e.g. flc) variants were not supported. Then I did a quick and dirty test by just renaming (without any format conversion) those files to supported format extensions (e.g. "mv test.dsf test_dsf.flac") and the result of ffmpegfs ("test_dsf.mp3") was usable i.e. its transconding worked from DSF to MP3.

I looked at the code (https://github.com/nschlia/ffmpegfs/blob/master/src/fuseops.cc#L382). I got an impression that "av_guess_format" is not very good for detection (I think it's mainly for ffmpeg supported output formats). I don't have really good idea how to improve the situation but could you consider implementing "-o"-option to provide a "whitelisted" file extensions that would be supported additionally.

**Comment by nschlia** [comment]
Thanks, great to hear it works as expected. I put a lot of efforts in that :)

Concerning the input format problem, yes, av_guess_format is very dumb. If you take a look at the code  (see https://ffmpeg.org/doxygen/3.1/format_8c_source.html#l00098), it simply compares the file extension.

There are performance reasons why I am using it. To find out if a file is really supported it would be required to open it, and scan it partially. That would be an overkill for bigger numbers of files.

A whitelist option would be a possible work around.

**Comment by kimmot** [comment]
I tested the allow list implementation for the supplementary audio file extensions with the attached patch. I only tested it very little so far but it seemed to work quite well. The comma delimiter for the extensions was not good choise for "-o audioextensions=...". What do you think about this approach?
[allow_list_audio_extensions.patch.txt](https://github.com/nschlia/ffmpegfs/files/7697612/allow_list_audio_extensions.patch.txt)

**Reply by nschlia** [reply]
Looks good, I'll probably accept it. I am thinking about extending it to all formats (video and audio alike). But I am a bit indisposed at the moment, my cat bit my left hand and it is currently bandaged. A bit cumbersome when typing :)
 So I am with you, but it might take awhile.
